front/components/assistant/conversation/input_bar/useConversationDrafts.ts

```diff
@@ -23,29 +24,44 @@ interface DraftStorage {
 const STORAGE_KEY = "conversation-drafts";
 const DRAFT_EXPIRY_MS = 7 * 24 * 60 * 60 * 1000;
 
+const getHasConversationId = (
+  conversationId: string | string[] | undefined
+): conversationId is string => {
+  if (conversationId === undefined || typeof conversationId !== "string") {
+    return false;
+  }
+
+  return true;
+};
+
 /**
  * Hook to manage per-conversation input drafts in localStorage.
  * Provides auto-save, retrieval, and cleanup functionality.
  */
 export function useConversationDrafts({
   workspaceId,
   userId,
-  conversationId,
 }: {
   workspaceId: string;
   userId: string | null;
-  conversationId: string | null;
 }) {
-  const internalConversationId = conversationId ?? "new";
-  const internalUserId = userId ?? "anonymous";
+  const router = useRouter();
+  // this will be undefined if you are in Agent Builder
+  const conversationId = router.query.cId;
```

Comment:
we need to get a value from router

Reply:
That's doomed to hit us back in the future. This might change, other pages could have a `cId`
Also this is not consistent as it's the sole parameter we get "magically" instead from the parameters, it makes testing this hook harder

I would rather add a if in the inputBar to _not_ save/restore the draft when we are in the agent builder